internal/releaseapi/client.go

```diff
@@ -118,12 +162,34 @@ func (c *Client) downloadBuild(build Build) (string, error) {
 	log.Printf("dowloading release archive from %s", build.URL)
 
 	zipFile, zipLength, err := c.downloadReleaseArchive(build)
+	defer os.Remove(zipFile.Name())
+	defer zipFile.Close()
 
 	if err != nil {
 		return "", err
 	}
 
-	defer zipFile.Close()
+	f, err := os.Open(zipFile.Name())
+	if err != nil {
+		return "", errors.Wrap(err, "could not open zip archive")
+	}
+	defer f.Close()
+
+	h := sha256.New()
+	if _, err := io.Copy(h, f); err != nil {
+		return "", errors.Wrap(err, "could not check sha256sum for zip archive")
+	}
+	sha256Sum := h.Sum(nil)
+
+	if checkSha256Sum != "" {
```

Comment:
I'm not sure when this would be the empty string - does HashiCorp occasionally not include checksums for certain builds?